pkgs/by-name/ca/cables/package.nix

```
@@ -0,0 +1,38 @@
+{ pkgs, lib, ... }:
+
+let
+  pname = "cables";
+  version = "0.3.2";
+  name = "${pname}-${version}";
+
+  src = pkgs.fetchurl {
+    url = "https://github.com/cables-gl/cables_electron/releases/download/v0.3.2/cables-0.3.2-linux-x64.AppImage";
+    sha256 = "sha256-D5Bgg5D03FPQF2HKow4yehcqToo4dpPudBg0W4UnqFs=";
+  };
+
+  appimageContents = pkgs.appimageTools.extract {
+    inherit pname version src;
+    postExtract = ''
+      substituteInPlace $out/${name}.desktop --replace 'Exec=AppRun' 'Exec=${pname}'
+    '';
+  };
+
+in
+pkgs.appimageTools.wrapType2 {
+  inherit name src;
+
+  extraInstallCommands = ''
+    mv $out/bin/${name} $out/bin/${pname}
+
+    install -m 444 -D ${appimageContents}/${name}.desktop $out/share/applications/${pname}.desktop
+    install -m 444 -D ${appimageContents}/${name}.png $out/share/icons/hicolor/512x512/apps/${pname}.png
+  '';
+
+  meta = with lib; {
+    description = "Standalone version of cables, a tool for creating beautiful interactive content.";
+    homepage = "https://cables.gl/standalone";
+    license = licenses.mit;
+    maintainers = with maintainers; [ rubikcubed ];
+    platforms = [ "x86_64-linux" ];
```

Platforms are meant to show which platforms the software supports. If we don't support it in nixpkgs, we mark it as broken (so someone can come and try to fix it if they want to). Make sure to add `stdenv` to the arguments up top.

```suggestion
    platforms = with platforms; linux ++ darwin ++ windows;
    broken = !(stdenv.hostPlatform.isLinux && stdenv.hostPlatform.isx86_64);
```

---

@h7x4 just to confirm, even if the package doesn't run outside linux, we still list them all in platforms if the software/website says it supports other platforms?

---

If it's our fault (in the extended sense that either something doesn't work, or we just didn't care to package and test it for other platforms, or nix just doesn't support it yet like windows), then we put them in `platforms` to say that the upstream software supports it, but also list it in the `broken` field to say that we don't support it (yet).

In this case, there are a bunch of extra download links for all popular platforms but we haven't packaged any other than x86_64 linux. And that's fine, it's certainly better than nothing :)